pkg/api/group.go

```diff
@@ -25,36 +25,44 @@ type GroupRepository interface {
 	List(ctx context.Context) ([]*model.Group, error)
 }
 
+type GroupAdder interface {
+	AddGroup(g *model.Group) error
+}
+
 type GroupService struct {
-	logger  logrus.FieldLogger
-	persons PersonLister
-	groups  GroupRepository
+	logger     logrus.FieldLogger
+	persons    PersonLister
+	groups     GroupRepository
+	groupAdder GroupAdder
 }
 
-func NewGroupService(lister PersonLister, groups GroupRepository, logger logrus.FieldLogger) (*GroupService, error) {
+func NewGroupService(lister PersonLister, groups GroupRepository, adder GroupAdder, logger logrus.FieldLogger) (*GroupService, error) {
 	if lister == nil || groups == nil {
 		return nil, ErrRepoNil
 	}
 	if logger == nil {
 		logger = logrus.StandardLogger()
 	}
 	return &GroupService{
-		logger:  logger.WithField("service", "group"),
-		persons: lister,
-		groups:  groups,
+		logger:     logger.WithField("service", "group"),
+		persons:    lister,
+		groups:     groups,
+		groupAdder: adder,
 	}, nil
 }
 
 func (s GroupService) List(w http.ResponseWriter, r *http.Request) {
 	logger := s.logger.WithField("method", "list")
 	groups, err := s.groups.List(r.Context())
 	if err != nil {
+		logger.Error(err)
```

Comment:
i think we should return the error and use logging middleware to indicate errors.

Reply:
Do you mean this specific error or every one? Also returning the error means we have to wrap every function in some error handler or use some custom type for the function which implements `http.Handler`.

Reply:
That's right. All endpoints/handler that return an error then need this log middleware.

Reply:
Alright sounds good to me, but that should be a separate pull request IMO.

Reply:
ok